pylint/pyreverse/utils.py

```diff
@@ -218,7 +218,7 @@ def infer_node(node: nodes.AssignAttr | nodes.AssignName) -> set[Any]:
     ann = get_annotation(node)
     try:
         if ann:
-            if isinstance(ann, nodes.Subscript):
+            if isinstance(ann, (nodes.Subscript, nodes.BinOp)):
```

Comment:
Perhaps we could make it more strict as this would print something like `int + str` to the diagram if a `+` is used in the type annotation.